From 5bbc557eea0641a9e6f98da64a66f864a925fa9b Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Fri, 25 Mar 2022 11:55:29 +0300
Subject: [PATCH 25/92] irqchip: renesas-intswcd: add full interrupt table for
 S4

Since interrupt ids are part of ABI between kernel and dtb, incremental
addition of interrupt table entries could lead to binary compatibility
issues over time.

To avoid it, populate the table completely, by adding all interrupts
supported by hardware.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 drivers/irqchip/irq-r8a779f0-intswcd.c        | 480 ++++++++++++++++-
 .../renesas-intswcd-r8a779f0.h                | 482 +++++++++++++++++-
 2 files changed, 949 insertions(+), 13 deletions(-)

diff --git a/drivers/irqchip/irq-r8a779f0-intswcd.c b/drivers/irqchip/irq-r8a779f0-intswcd.c
index 230358fdbb30..d7cf421146f7 100644
--- a/drivers/irqchip/irq-r8a779f0-intswcd.c
+++ b/drivers/irqchip/irq-r8a779f0-intswcd.c
@@ -13,11 +13,481 @@ static const struct intswcd_hwirq_info r8a779f0_ihi[] = {
 		.mask = BIT(_bit),			\
 	},
 
-	ENTRY(INTRIIC0EE,  LEVEL, 33,  9, 0)
-	ENTRY(INTRIIC0TEI, LEVEL, 33,  9, 1)
-	ENTRY(INTRIIC0RI,   EDGE, 24, 13, 0)
-	ENTRY(INTRIIC0TI,   EDGE, 24, 13, 1)
-	ENTRY(INTOSTM0TINT, EDGE, 12,  2, 0)
+	ENTRY(CAN0TSOW0,	LEVEL,	 0, 13,  0)
+	ENTRY(CAN0TSOW1,	LEVEL,	 0, 13,  1)
+	ENTRY(CAN0TSOW2,	LEVEL,	 0, 13,  2)
+	ENTRY(CAN0TSOW3,	LEVEL,	 0, 13,  3)
+	ENTRY(CAN0TSOW4,	LEVEL,	 0, 13,  4)
+	ENTRY(CAN0TSOW5,	LEVEL,	 0, 13,  5)
+	ENTRY(CAN0TSOW6,	LEVEL,	 0, 13,  6)
+	ENTRY(CAN0TSOW7,	LEVEL,	 0, 13,  7)
+	ENTRY(CAN0TSIDC0,	LEVEL,	 0, 13,  8)
+	ENTRY(CAN0TSIDC1,	LEVEL,	 0, 13,  9)
+	ENTRY(CAN0TSIDC2,	LEVEL,	 0, 13, 10)
+	ENTRY(CAN0TSIDC3,	LEVEL,	 0, 13, 11)
+	ENTRY(CAN0TSIDC4,	LEVEL,	 0, 13, 12)
+	ENTRY(CAN0TSIDC5,	LEVEL,	 0, 13, 13)
+	ENTRY(CAN0TSIDC6,	LEVEL,	 0, 13, 14)
+	ENTRY(CAN0TSIDC7,	LEVEL,	 0, 13, 15)
+
+	ENTRY(PE0_INTEINTSW0,	LEVEL,	 1,  0, 14)
+	ENTRY(PE0_INTEINTSW1,	LEVEL,	 1,  0, 15)
+	ENTRY(PE0_INTEINTSW2,	LEVEL,	 1,  0, 16)
+	ENTRY(PE0_INTEINTSW3,	LEVEL,	 1,  0, 17)
+
+	ENTRY(INTICU2PES0,	LEVEL,	 2,  0, 20)
+	ENTRY(INTICU2PE0,	LEVEL,	 2,  0, 21)
+
+	ENTRY(PE0_INTTPTM0,	LEVEL,	 3,  0, 31)
+
+	ENTRY(CAN1TSOW0,	LEVEL,	 4, 13, 16)
+	ENTRY(CAN1TSOW1,	LEVEL,	 4, 13, 17)
+	ENTRY(CAN1TSOW2,	LEVEL,	 4, 13, 18)
+	ENTRY(CAN1TSOW3,	LEVEL,	 4, 13, 19)
+	ENTRY(CAN1TSOW4,	LEVEL,	 4, 13, 20)
+	ENTRY(CAN1TSOW5,	LEVEL,	 4, 13, 21)
+	ENTRY(CAN1TSOW6,	LEVEL,	 4, 13, 22)
+	ENTRY(CAN1TSOW7,	LEVEL,	 4, 13, 23)
+	ENTRY(CAN1TSIDC0,	LEVEL,	 4, 13, 24)
+	ENTRY(CAN1TSIDC1,	LEVEL,	 4, 13, 25)
+	ENTRY(CAN1TSIDC2,	LEVEL,	 4, 13, 26)
+	ENTRY(CAN1TSIDC3,	LEVEL,	 4, 13, 27)
+	ENTRY(CAN1TSIDC4,	LEVEL,	 4, 13, 28)
+	ENTRY(CAN1TSIDC5,	LEVEL,	 4, 13, 29)
+	ENTRY(CAN1TSIDC6,	LEVEL,	 4, 13, 30)
+	ENTRY(CAN1TSIDC7,	LEVEL,	 4, 13, 31)
+
+	ENTRY(INTICU2PES1,	LEVEL,	 6,  1, 20)
+	ENTRY(INTICU2PE1,	LEVEL,	 6,  1, 21)
+
+	ENTRY(PE1_INTTPTM1,	LEVEL,	 7,  1, 31)
+
+	ENTRY(INTDTS31TO0,	LEVEL,	 8,  2,  0)
+	ENTRY(INTDTS63TO32,	LEVEL,	 8,  2,  1)
+	ENTRY(INTDTS95TO64,	LEVEL,	 8,  2,  2)
+	ENTRY(INTDTS127TO96,	LEVEL,	 8,  2,  3)
+	ENTRY(INTDTSCT31TO0,	LEVEL,	 8,  2,  3)
+	ENTRY(INTDTSCT63TO32,	LEVEL,	 8,  2,  4)
+	ENTRY(INTDTSCT95TO64,	LEVEL,	 8,  2,  6)
+	ENTRY(INTDTSCT127TO96,	LEVEL,	 8,  2,  7)
+
+	ENTRY(INTSDMAC0CH0,	LEVEL,	 9,  3,  0)
+	ENTRY(INTSDMAC0CH1,	LEVEL,	 9,  3,  1)
+	ENTRY(INTSDMAC0CH2,	LEVEL,	 9,  3,  2)
+	ENTRY(INTSDMAC0CH3,	LEVEL,	 9,  3,  3)
+	ENTRY(INTSDMAC0CH4,	LEVEL,	 9,  3,  4)
+	ENTRY(INTSDMAC0CH5,	LEVEL,	 9,  3,  5)
+	ENTRY(INTSDMAC0CH6,	LEVEL,	 9,  3,  6)
+	ENTRY(INTSDMAC0CH7,	LEVEL,	 9,  3,  7)
+	ENTRY(INTSDMAC0CH8,	LEVEL,	 9,  3,  8)
+	ENTRY(INTSDMAC0CH9,	LEVEL,	 9,  3,  9)
+	ENTRY(INTSDMAC0CH10,	LEVEL,	 9,  3, 10)
+	ENTRY(INTSDMAC0CH11,	LEVEL,	 9,  3, 11)
+	ENTRY(INTSDMAC0CH12,	LEVEL,	 9,  3, 12)
+	ENTRY(INTSDMAC0CH13,	LEVEL,	 9,  3, 13)
+	ENTRY(INTSDMAC0CH14,	LEVEL,	 9,  3, 14)
+	ENTRY(INTSDMAC0CH15,	LEVEL,	 9,  3, 15)
+
+	ENTRY(INTSDMAC1CH0,	LEVEL,	10,  3, 16)
+	ENTRY(INTSDMAC1CH1,	LEVEL,	10,  3, 17)
+	ENTRY(INTSDMAC1CH2,	LEVEL,	10,  3, 18)
+	ENTRY(INTSDMAC1CH3,	LEVEL,	10,  3, 19)
+	ENTRY(INTSDMAC1CH4,	LEVEL,	10,  3, 20)
+	ENTRY(INTSDMAC1CH5,	LEVEL,	10,  3, 21)
+	ENTRY(INTSDMAC1CH6,	LEVEL,	10,  3, 22)
+	ENTRY(INTSDMAC1CH7,	LEVEL,	10,  3, 23)
+	ENTRY(INTSDMAC1CH8,	LEVEL,	10,  3, 24)
+	ENTRY(INTSDMAC1CH9,	LEVEL,	10,  3, 25)
+	ENTRY(INTSDMAC1CH10,	LEVEL,	10,  3, 26)
+	ENTRY(INTSDMAC1CH11,	LEVEL,	10,  3, 27)
+	ENTRY(INTSDMAC1CH12,	LEVEL,	10,  3, 28)
+	ENTRY(INTSDMAC1CH13,	LEVEL,	10,  3, 29)
+	ENTRY(INTSDMAC1CH14,	LEVEL,	10,  3, 30)
+	ENTRY(INTSDMAC1CH15,	LEVEL,	10,  3, 31)
+
+	ENTRY(INTRCANGERR0,	LEVEL,	12,  4,  0)
+	ENTRY(INTRCANGRECC0,	LEVEL,	12,  4,  1)
+	ENTRY(INTRCAN0ERR,	LEVEL,	13,  4,  2)
+	ENTRY(INTRCAN0REC,	LEVEL,	13,  4,  3)
+	ENTRY(INTRCAN0TRX,	LEVEL,	13,  4,  4)
+	ENTRY(INTRCAN1ERR,	LEVEL,	14,  4,  5)
+	ENTRY(INTRCAN1REC,	LEVEL,	14,  4,  6)
+	ENTRY(INTRCAN1TRX,	LEVEL,	14,  4,  7)
+	ENTRY(INTRCAN2ERR,	LEVEL,	15,  4,  8)
+	ENTRY(INTRCAN2REC,	LEVEL,	15,  4,  9)
+	ENTRY(INTRCAN2TRX,	LEVEL,	15,  4, 10)
+	ENTRY(INTRCAN3ERR,	LEVEL,	16,  4, 11)
+	ENTRY(INTRCAN3REC,	LEVEL,	16,  4, 12)
+	ENTRY(INTRCAN3TRX,	LEVEL,	16,  4, 13)
+	ENTRY(INTRCAN4ERR,	LEVEL,	17,  4, 14)
+	ENTRY(INTRCAN4REC,	LEVEL,	17,  4, 15)
+	ENTRY(INTRCAN4TRX,	LEVEL,	17,  4, 16)
+	ENTRY(INTRCAN5ERR,	LEVEL,	18,  4, 17)
+	ENTRY(INTRCAN5REC,	LEVEL,	18,  4, 18)
+	ENTRY(INTRCAN5TRX,	LEVEL,	18,  4, 19)
+	ENTRY(INTRCAN6ERR,	LEVEL,	19,  4, 20)
+	ENTRY(INTRCAN6REC,	LEVEL,	19,  4, 21)
+	ENTRY(INTRCAN6TRX,	LEVEL,	19,  4, 22)
+	ENTRY(INTRCAN7ERR,	LEVEL,	20,  4, 23)
+	ENTRY(INTRCAN7REC,	LEVEL,	20,  4, 24)
+	ENTRY(INTRCAN7TRX,	LEVEL,	20,  4, 25)
+
+	ENTRY(INTRCANGERR1,	LEVEL,	21,  5,  0)
+	ENTRY(INTRCANGRECC1,	LEVEL,	21,  5,  1)
+	ENTRY(INTRCAN8ERR,	LEVEL,	22,  5,  2)
+	ENTRY(INTRCAN8REC,	LEVEL,	22,  5,  3)
+	ENTRY(INTRCAN8TRX,	LEVEL,	22,  5,  4)
+	ENTRY(INTRCAN9ERR,	LEVEL,	23,  5,  5)
+	ENTRY(INTRCAN9REC,	LEVEL,	23,  5,  6)
+	ENTRY(INTRCAN9TRX,	LEVEL,	23,  5,  7)
+	ENTRY(INTRCAN10ERR,	LEVEL,	24,  5,  8)
+	ENTRY(INTRCAN10REC,	LEVEL,	24,  5,  9)
+	ENTRY(INTRCAN10TRX,	LEVEL,	24,  5, 10)
+	ENTRY(INTRCAN11ERR,	LEVEL,	25,  5, 11)
+	ENTRY(INTRCAN11REC,	LEVEL,	25,  5, 12)
+	ENTRY(INTRCAN11TRX,	LEVEL,	25,  5, 13)
+	ENTRY(INTRCAN12ERR,	LEVEL,	26,  5, 14)
+	ENTRY(INTRCAN12REC,	LEVEL,	26,  5, 15)
+	ENTRY(INTRCAN12TRX,	LEVEL,	26,  5, 16)
+	ENTRY(INTRCAN13ERR,	LEVEL,	27,  5, 17)
+	ENTRY(INTRCAN13REC,	LEVEL,	27,  5, 18)
+	ENTRY(INTRCAN13TRX,	LEVEL,	27,  5, 19)
+	ENTRY(INTRCAN14ERR,	LEVEL,	28,  5, 20)
+	ENTRY(INTRCAN14REC,	LEVEL,	28,  5, 21)
+	ENTRY(INTRCAN14TRX,	LEVEL,	28,  5, 22)
+	ENTRY(INTRCAN15ERR,	LEVEL,	29,  5, 23)
+	ENTRY(INTRCAN15REC,	LEVEL,	29,  5, 24)
+	ENTRY(INTRCAN15TRX,	LEVEL,	29,  5, 25)
+
+	ENTRY(INTRSENT0SI,	LEVEL,	30,  6,  0)
+	ENTRY(INTRSENT1SI,	LEVEL,	30,  6,  1)
+	ENTRY(INTRSENT2SI,	LEVEL,	30,  6,  2)
+	ENTRY(INTRSENT3SI,	LEVEL,	30,  6,  3)
+	ENTRY(INTRSENT4SI,	LEVEL,	30,  6,  4)
+	ENTRY(INTRSENT5SI,	LEVEL,	30,  6,  5)
+	ENTRY(INTRSENT6SI,	LEVEL,	30,  6,  6)
+	ENTRY(INTRSENT7SI,	LEVEL,	30,  6,  7)
+
+	ENTRY(INTFXLA0LINE0,	LEVEL,	31,  7,  0)
+	ENTRY(INTFXLA0LINE1,	LEVEL,	31,  7,  1)
+	ENTRY(INTFXLA0TIM0,	LEVEL,	31,  7,  2)
+	ENTRY(INTFXLA0TIM1,	LEVEL,	31,  7,  3)
+	ENTRY(INTFXLA0TIM2,	LEVEL,	31,  7,  4)
+	ENTRY(INTFXLA0FDA,	LEVEL,	31,  7,  5)
+	ENTRY(INTFXLA0FW,	LEVEL,	31,  7,  6)
+	ENTRY(INTFXLA0OW,	LEVEL,	31,  7,  7)
+	ENTRY(INTFXLA0OT,	LEVEL,	31,  7,  8)
+	ENTRY(INTFXLA0IQF,	LEVEL,	31,  7,  9)
+	ENTRY(INTFXLA0IQE,	LEVEL,	31,  7, 10)
+
+	ENTRY(INTETNB0DATA,	LEVEL,	32,  8,  0)
+	ENTRY(INTETNB0ERR,	LEVEL,	32,  8,  1)
+	ENTRY(INTETNB0MNG,	LEVEL,	32,  8,  2)
+	ENTRY(INTETNB0MAC,	LEVEL,	32,  8,  3)
+
+	ENTRY(INTRIIC0EE,  	LEVEL,	33,  9,  0)
+	ENTRY(INTRIIC0TEI, 	LEVEL,	33,  9,  1)
+
+	ENTRY(RCAN0VM0TX, 	LEVEL,	35, 11,  0)
+	ENTRY(RCAN0VM0RX, 	LEVEL,	35, 11,  8)
+	ENTRY(RCAN0VM0ERR, 	LEVEL,	35, 11, 16)
+	ENTRY(RCAN0VM1TX, 	LEVEL,	36, 11,  1)
+	ENTRY(RCAN0VM1RX, 	LEVEL,	36, 11,  9)
+	ENTRY(RCAN0VM1ERR, 	LEVEL,	36, 11, 17)
+	ENTRY(RCAN0VM2TX, 	LEVEL,	37, 11,  2)
+	ENTRY(RCAN0VM2RX, 	LEVEL,	37, 11, 10)
+	ENTRY(RCAN0VM2ERR, 	LEVEL,	37, 11, 18)
+	ENTRY(RCAN0VM3TX, 	LEVEL,	38, 11,  3)
+	ENTRY(RCAN0VM3RX, 	LEVEL,	38, 11, 11)
+	ENTRY(RCAN0VM3ERR, 	LEVEL,	38, 11, 19)
+	ENTRY(RCAN0VM4TX, 	LEVEL,	39, 11,  4)
+	ENTRY(RCAN0VM4RX, 	LEVEL,	39, 11, 12)
+	ENTRY(RCAN0VM4ERR, 	LEVEL,	39, 11, 20)
+	ENTRY(RCAN0VM5TX, 	LEVEL,	40, 11,  5)
+	ENTRY(RCAN0VM5RX, 	LEVEL,	40, 11, 13)
+	ENTRY(RCAN0VM5ERR, 	LEVEL,	40, 11, 21)
+	ENTRY(RCAN0VM6TX, 	LEVEL,	41, 11,  6)
+	ENTRY(RCAN0VM6RX, 	LEVEL,	41, 11, 14)
+	ENTRY(RCAN0VM6ERR, 	LEVEL,	41, 11, 22)
+	ENTRY(RCAN0VM7TX, 	LEVEL,	42, 11,  7)
+	ENTRY(RCAN0VM7RX, 	LEVEL,	42, 11, 15)
+	ENTRY(RCAN0VM7ERR, 	LEVEL,	42, 11, 23)
+
+	ENTRY(RCAN1VM0TX, 	LEVEL,	43, 12,  0)
+	ENTRY(RCAN1VM0RX, 	LEVEL,	43, 12,  8)
+	ENTRY(RCAN1VM0ERR, 	LEVEL,	43, 12, 16)
+	ENTRY(RCAN1VM1TX, 	LEVEL,	44, 12,  1)
+	ENTRY(RCAN1VM1RX, 	LEVEL,	44, 12,  9)
+	ENTRY(RCAN1VM1ERR, 	LEVEL,	44, 12, 17)
+	ENTRY(RCAN1VM2TX, 	LEVEL,	45, 12,  2)
+	ENTRY(RCAN1VM2RX, 	LEVEL,	45, 12, 10)
+	ENTRY(RCAN1VM2ERR, 	LEVEL,	45, 12, 18)
+	ENTRY(RCAN1VM3TX, 	LEVEL,	46, 12,  3)
+	ENTRY(RCAN1VM3RX, 	LEVEL,	46, 12, 11)
+	ENTRY(RCAN1VM3ERR, 	LEVEL,	46, 12, 19)
+	ENTRY(RCAN1VM4TX, 	LEVEL,	47, 12,  4)
+	ENTRY(RCAN1VM4RX, 	LEVEL,	47, 12, 12)
+	ENTRY(RCAN1VM4ERR, 	LEVEL,	47, 12, 20)
+	ENTRY(RCAN1VM5TX, 	LEVEL,	48, 12,  5)
+	ENTRY(RCAN1VM5RX, 	LEVEL,	48, 12, 13)
+	ENTRY(RCAN1VM5ERR, 	LEVEL,	48, 12, 21)
+	ENTRY(RCAN1VM6TX, 	LEVEL,	49, 12,  6)
+	ENTRY(RCAN1VM6RX, 	LEVEL,	49, 12, 14)
+	ENTRY(RCAN1VM6ERR, 	LEVEL,	49, 12, 22)
+	ENTRY(RCAN1VM7TX, 	LEVEL,	50, 12,  7)
+	ENTRY(RCAN1VM7RX, 	LEVEL,	50, 12, 15)
+	ENTRY(RCAN1VM7ERR, 	LEVEL,	50, 12, 23)
+
+	ENTRY(PE0_INTECMMI,	EDGE,	 1,  0,  8)
+	ENTRY(PE0_INTECDCLSMI,	EDGE,	 1,  0,  9)
+
+	ENTRY(INTTAUD0I0,	EDGE,	 2,  0, 10)
+	ENTRY(INTTAUD0I2,	EDGE,	 2,  0, 11)
+	ENTRY(INTTAUD0I4,	EDGE,	 2,  0, 12)
+	ENTRY(INTTAUD0I6,	EDGE,	 2,  0, 13)
+
+	ENTRY(INTWDTB0TIT,	EDGE,	 3,  0, 22)
+
+	ENTRY(INTTAUD0I8,	EDGE,	 4,  0, 23)
+	ENTRY(INTTAUD0I10,	EDGE,	 4,  0, 24)
+	ENTRY(INTTAUD0I12,	EDGE,	 4,  0, 25)
+	ENTRY(INTTAUD0I14,	EDGE,	 4,  0, 26)
+
+	ENTRY(INTSDMACERR,	EDGE,	 5,  0, 29)
+
+	ENTRY(INTDTSERR,	EDGE,	 5,  0, 30)
+
+	ENTRY(PE1_INTECMMI,	EDGE,	 7,  1,  8)
+	ENTRY(PE1_INTECDCLSMI,	EDGE,	 7,  1,  9)
+
+	ENTRY(INTWDTB1TIT,	EDGE,	 9,  1, 22)
+
+	ENTRY(INTOSTM0TINT,	EDGE,	12,  2,  0)
+	ENTRY(INTOSTM1TINT,	EDGE,	12,  2,  1)
+	ENTRY(INTOSTM2TINT,	EDGE,	12,  2,  2)
+	ENTRY(INTOSTM3TINT,	EDGE,	12,  2,  3)
+	ENTRY(INTOSTM4TINT,	EDGE,	12,  2,  4)
+	ENTRY(INTOSTM5TINT,	EDGE,	12,  2,  5)
+	ENTRY(INTOSTM8TINT,	EDGE,	12,  2,  6)
+	ENTRY(INTOSTM9TINT,	EDGE,	12,  2,  7)
+
+	ENTRY(INTTPTMU00,	EDGE,	13,  2,  8)
+	ENTRY(INTTPTMU01,	EDGE,	13,  2,  9)
+	ENTRY(INTTPTMU02,	EDGE,	13,  2, 10)
+	ENTRY(INTTPTMU03,	EDGE,	13,  2, 11)
+	ENTRY(INTTPTMU10,	EDGE,	13,  2, 12)
+	ENTRY(INTTPTMU11,	EDGE,	13,  2, 13)
+	ENTRY(INTTPTMU12,	EDGE,	13,  2, 14)
+	ENTRY(INTTPTMU13,	EDGE,	13,  2, 15)
+
+	ENTRY(INTMSPI0TX0,	EDGE,	14,  3,  0)
+	ENTRY(INTMSPI0RX0,	EDGE,	14,  3,  1)
+	ENTRY(INTMSPI0TX1,	EDGE,	14,  3,  2)
+	ENTRY(INTMSPI0RX1,	EDGE,	14,  3,  3)
+	ENTRY(INTMSPI0TX2,	EDGE,	14,  3,  4)
+	ENTRY(INTMSPI0RX2,	EDGE,	14,  3,  5)
+	ENTRY(INTMSPI0TX,	EDGE,	15,  4,  0)
+	ENTRY(INTMSPI0RX,	EDGE,	15,  4,  1)
+	ENTRY(INTMSPI0FE,	EDGE,	15,  4,  2)
+	ENTRY(INTMSPI0ERR,	EDGE,	15,  4,  3)
+
+	ENTRY(INTMSPI1TX0,	EDGE,	15,  4,  4)
+	ENTRY(INTMSPI1RX0,	EDGE,	15,  4,  5)
+	ENTRY(INTMSPI1TX1,	EDGE,	15,  4,  6)
+	ENTRY(INTMSPI1RX1,	EDGE,	15,  4,  7)
+	ENTRY(INTMSPI1TX2,	EDGE,	15,  4,  8)
+	ENTRY(INTMSPI1RX2,	EDGE,	15,  4,  9)
+	ENTRY(INTMSPI1TX,	EDGE,	15,  4, 10)
+	ENTRY(INTMSPI1RX,	EDGE,	15,  4, 11)
+	ENTRY(INTMSPI1FE,	EDGE,	15,  4, 12)
+	ENTRY(INTMSPI1ERR,	EDGE,	15,  4, 13)
+
+	ENTRY(INTMSPI2TX,	EDGE,	15,  4, 14)
+	ENTRY(INTMSPI2RX,	EDGE,	15,  4, 15)
+	ENTRY(INTMSPI2FE,	EDGE,	15,  4, 16)
+	ENTRY(INTMSPI2ERR,	EDGE,	15,  4, 17)
+
+	ENTRY(INTMSPI3TX,	EDGE,	15,  4, 18)
+	ENTRY(INTMSPI3RX,	EDGE,	15,  4, 19)
+	ENTRY(INTMSPI3FE,	EDGE,	15,  4, 20)
+	ENTRY(INTMSPI3ERR,	EDGE,	15,  4, 21)
+
+	ENTRY(INTMSPI4TX,	EDGE,	15,  4, 22)
+	ENTRY(INTMSPI4RX,	EDGE,	15,  4, 23)
+	ENTRY(INTMSPI4FE,	EDGE,	15,  4, 24)
+	ENTRY(INTMSPI4ERR,	EDGE,	15,  4, 25)
+
+	ENTRY(INTMSPI5TX,	EDGE,	15,  4, 26)
+	ENTRY(INTMSPI5RX,	EDGE,	15,  4, 27)
+	ENTRY(INTMSPI5FE,	EDGE,	15,  4, 28)
+	ENTRY(INTMSPI5ERR,	EDGE,	15,  4, 29)
+
+	ENTRY(INTTAUJ1I0,	EDGE,	16,  5,  0)
+	ENTRY(INTTAUJ1I1,	EDGE,	16,  5,  1)
+	ENTRY(INTTAUJ1I2,	EDGE,	16,  5,  2)
+	ENTRY(INTTAUJ1I3,	EDGE,	16,  5,  3)
+	ENTRY(INTTAUJ3I0,	EDGE,	16,  5,  4)
+	ENTRY(INTTAUJ3I1,	EDGE,	16,  5,  5)
+	ENTRY(INTTAUJ3I2,	EDGE,	16,  5,  6)
+	ENTRY(INTTAUJ3I3,	EDGE,	16,  5,  7)
+
+	ENTRY(INTTAUD0I1,	EDGE,	17,  6,  0)
+	ENTRY(INTTAUD0I3,	EDGE,	17,  6,  1)
+	ENTRY(INTTAUD0I5,	EDGE,	17,  6,  2)
+	ENTRY(INTTAUD0I7,	EDGE,	17,  6,  3)
+	ENTRY(INTTAUD0I9,	EDGE,	17,  6,  4)
+	ENTRY(INTTAUD0I11,	EDGE,	17,  6,  5)
+	ENTRY(INTTAUD0I13,	EDGE,	17,  6,  6)
+	ENTRY(INTTAUD0I15,	EDGE,	17,  6,  7)
+	ENTRY(INTTAUD1I0,	EDGE,	17,  6,  8)
+	ENTRY(INTTAUD1I1,	EDGE,	17,  6,  9)
+	ENTRY(INTTAUD1I2,	EDGE,	17,  6, 10)
+	ENTRY(INTTAUD1I3,	EDGE,	17,  6, 11)
+	ENTRY(INTTAUD1I4,	EDGE,	17,  6, 12)
+	ENTRY(INTTAUD1I5,	EDGE,	17,  6, 13)
+	ENTRY(INTTAUD1I6,	EDGE,	17,  6, 14)
+	ENTRY(INTTAUD1I7,	EDGE,	17,  6, 15)
+	ENTRY(INTTAUD1I8,	EDGE,	17,  6, 16)
+	ENTRY(INTTAUD1I9,	EDGE,	17,  6, 17)
+	ENTRY(INTTAUD1I10,	EDGE,	17,  6, 18)
+	ENTRY(INTTAUD1I11,	EDGE,	17,  6, 19)
+	ENTRY(INTTAUD1I12,	EDGE,	17,  6, 20)
+	ENTRY(INTTAUD1I13,	EDGE,	17,  6, 21)
+	ENTRY(INTTAUD1I14,	EDGE,	17,  6, 22)
+	ENTRY(INTTAUD1I15,	EDGE,	17,  6, 23)
+
+	ENTRY(INTRLIN30,	EDGE,	18,  7,  0)
+	ENTRY(INTRLIN30UR0,	EDGE,	18,  7,  1)
+	ENTRY(INTRLIN30UR1,	EDGE,	18,  7,  2)
+	ENTRY(INTRLIN30UR2,	EDGE,	18,  7,  3)
+	ENTRY(INTRLIN31,	EDGE,	18,  7,  4)
+	ENTRY(INTRLIN31UR0,	EDGE,	18,  7,  5)
+	ENTRY(INTRLIN31UR1,	EDGE,	18,  7,  6)
+	ENTRY(INTRLIN31UR2,	EDGE,	18,  7,  7)
+	ENTRY(INTRLIN32,	EDGE,	18,  7,  8)
+	ENTRY(INTRLIN32UR0,	EDGE,	18,  7,  9)
+	ENTRY(INTRLIN32UR1,	EDGE,	18,  7, 10)
+	ENTRY(INTRLIN32UR2,	EDGE,	18,  7, 11)
+	ENTRY(INTRLIN33,	EDGE,	18,  7, 12)
+	ENTRY(INTRLIN33UR0,	EDGE,	18,  7, 13)
+	ENTRY(INTRLIN33UR1,	EDGE,	18,  7, 14)
+	ENTRY(INTRLIN33UR2,	EDGE,	18,  7, 15)
+	ENTRY(INTRLIN34,	EDGE,	18,  7, 16)
+	ENTRY(INTRLIN34UR0,	EDGE,	18,  7, 17)
+	ENTRY(INTRLIN34UR1,	EDGE,	18,  7, 18)
+	ENTRY(INTRLIN34UR2,	EDGE,	18,  7, 19)
+	ENTRY(INTRLIN35,	EDGE,	18,  7, 20)
+	ENTRY(INTRLIN35UR0,	EDGE,	18,  7, 21)
+	ENTRY(INTRLIN35UR1,	EDGE,	18,  7, 22)
+	ENTRY(INTRLIN35UR2,	EDGE,	18,  7, 23)
+	ENTRY(INTRLIN36,	EDGE,	18,  7, 24)
+	ENTRY(INTRLIN36UR0,	EDGE,	18,  7, 25)
+	ENTRY(INTRLIN36UR1,	EDGE,	18,  7, 26)
+	ENTRY(INTRLIN36UR2,	EDGE,	18,  7, 27)
+	ENTRY(INTRLIN37,	EDGE,	18,  7, 28)
+	ENTRY(INTRLIN37UR0,	EDGE,	18,  7, 29)
+	ENTRY(INTRLIN37UR1,	EDGE,	18,  7, 30)
+	ENTRY(INTRLIN37UR2,	EDGE,	18,  7, 31)
+
+	ENTRY(INTRLIN38,	EDGE,	19,  8,  0)
+	ENTRY(INTRLIN38UR0,	EDGE,	19,  8,  1)
+	ENTRY(INTRLIN38UR1,	EDGE,	19,  8,  2)
+	ENTRY(INTRLIN38UR2,	EDGE,	19,  8,  3)
+	ENTRY(INTRLIN39,	EDGE,	19,  8,  4)
+	ENTRY(INTRLIN39UR0,	EDGE,	19,  8,  5)
+	ENTRY(INTRLIN39UR1,	EDGE,	19,  8,  6)
+	ENTRY(INTRLIN39UR2,	EDGE,	19,  8,  7)
+	ENTRY(INTRLIN310,	EDGE,	19,  8,  8)
+	ENTRY(INTRLIN310UR0,	EDGE,	19,  8,  9)
+	ENTRY(INTRLIN310UR1,	EDGE,	19,  8, 10)
+	ENTRY(INTRLIN310UR2,	EDGE,	19,  8, 11)
+	ENTRY(INTRLIN311,	EDGE,	19,  8, 12)
+	ENTRY(INTRLIN311UR0,	EDGE,	19,  8, 13)
+	ENTRY(INTRLIN311UR1,	EDGE,	19,  8, 14)
+	ENTRY(INTRLIN311UR2,	EDGE,	19,  8, 15)
+	ENTRY(INTRLIN312,	EDGE,	19,  8, 16)
+	ENTRY(INTRLIN312UR0,	EDGE,	19,  8, 17)
+	ENTRY(INTRLIN312UR1,	EDGE,	19,  8, 18)
+	ENTRY(INTRLIN312UR2,	EDGE,	19,  8, 19)
+	ENTRY(INTRLIN313,	EDGE,	19,  8, 20)
+	ENTRY(INTRLIN313UR0,	EDGE,	19,  8, 21)
+	ENTRY(INTRLIN313UR1,	EDGE,	19,  8, 22)
+	ENTRY(INTRLIN313UR2,	EDGE,	19,  8, 23)
+	ENTRY(INTRLIN314,	EDGE,	19,  8, 24)
+	ENTRY(INTRLIN314UR0,	EDGE,	19,  8, 25)
+	ENTRY(INTRLIN314UR1,	EDGE,	19,  8, 26)
+	ENTRY(INTRLIN314UR2,	EDGE,	19,  8, 27)
+	ENTRY(INTRLIN315,	EDGE,	19,  8, 28)
+	ENTRY(INTRLIN315UR0,	EDGE,	19,  8, 29)
+	ENTRY(INTRLIN315UR1,	EDGE,	19,  8, 30)
+	ENTRY(INTRLIN315UR2,	EDGE,	19,  8, 31)
+
+	ENTRY(INTRSENT0RI,	EDGE,	20,  9,  0)
+	ENTRY(INTRSENT1RI,	EDGE,	20,  9,  1)
+	ENTRY(INTRSENT2RI,	EDGE,	20,  9,  2)
+	ENTRY(INTRSENT3RI,	EDGE,	20,  9,  3)
+	ENTRY(INTRSENT4RI,	EDGE,	20,  9,  4)
+	ENTRY(INTRSENT5RI,	EDGE,	20,  9,  5)
+	ENTRY(INTRSENT6RI,	EDGE,	20,  9,  6)
+	ENTRY(INTRSENT7RI,	EDGE,	20,  9,  7)
+
+	ENTRY(INTP00,		EDGE,	21, 10,  0)
+	ENTRY(INTP01,		EDGE,	21, 10,  1)
+	ENTRY(INTP02,		EDGE,	21, 10,  2)
+	ENTRY(INTP03,		EDGE,	21, 10,  3)
+	ENTRY(INTP04,		EDGE,	21, 10,  4)
+	ENTRY(INTP05,		EDGE,	21, 10,  5)
+	ENTRY(INTP06,		EDGE,	21, 10,  6)
+	ENTRY(INTP07,		EDGE,	21, 10,  7)
+	ENTRY(INTP08,		EDGE,	21, 10,  8)
+	ENTRY(INTP09,		EDGE,	21, 10,  9)
+	ENTRY(INTP10,		EDGE,	21, 10, 10)
+	ENTRY(INTP11,		EDGE,	21, 10, 11)
+	ENTRY(INTP12,		EDGE,	21, 10, 12)
+	ENTRY(INTP13,		EDGE,	21, 10, 13)
+	ENTRY(INTP14,		EDGE,	21, 10, 14)
+	ENTRY(INTP15,		EDGE,	21, 10, 15)
+	ENTRY(INTP16,		EDGE,	21, 10, 16)
+	ENTRY(INTP17,		EDGE,	21, 10, 17)
+	ENTRY(INTP18,		EDGE,	21, 10, 18)
+	ENTRY(INTP19,		EDGE,	21, 10, 19)
+	ENTRY(INTP20,		EDGE,	21, 10, 20)
+	ENTRY(INTP21,		EDGE,	21, 10, 21)
+	ENTRY(INTP22,		EDGE,	21, 10, 22)
+	ENTRY(INTP23,		EDGE,	21, 10, 23)
+	ENTRY(INTP24,		EDGE,	21, 10, 24)
+	ENTRY(INTP25,		EDGE,	21, 10, 25)
+	ENTRY(INTP26,		EDGE,	21, 10, 26)
+	ENTRY(INTP27,		EDGE,	21, 10, 27)
+	ENTRY(INTP28,		EDGE,	21, 10, 28)
+	ENTRY(INTP29,		EDGE,	21, 10, 29)
+	ENTRY(INTP30,		EDGE,	21, 10, 30)
+	ENTRY(INTP31,		EDGE,	21, 10, 31)
+	ENTRY(INTP32,		EDGE,	22, 11,  0)
+	ENTRY(INTP33,		EDGE,	22, 11,  1)
+	ENTRY(INTP34,		EDGE,	22, 11,  2)
+	ENTRY(INTP35,		EDGE,	22, 11,  3)
+	ENTRY(INTP36,		EDGE,	22, 11,  4)
+	ENTRY(INTP37,		EDGE,	22, 11,  5)
+
+	ENTRY(INTRTCA01S,	EDGE,	23, 12,  0)
+	ENTRY(INTRTCA0AL,	EDGE,	23, 12,  1)
+	ENTRY(INTRTCA0R,	EDGE,	23, 12,  2)
+
+	ENTRY(INTRIIC0RI,	EDGE,	24, 13,  0)
+	ENTRY(INTRIIC0TI,	EDGE,	24, 13,  1)
+
+	ENTRY(INTDNFA1WUF0,	EDGE,	25, 14,  0)
+	ENTRY(INTDNFA1WUF1,	EDGE,	25, 14,  1)
+	ENTRY(INTDNFA2WUF0,	EDGE,	25, 14,  2)
+	ENTRY(INTDNFA2WUF1,	EDGE,	25, 14,  3)
+	ENTRY(INTDNFA0WUF0,	EDGE,	25, 14,  4)
+
+	ENTRY(INTICUMFATALERR,	EDGE,	26, 15,  0)
+
+	ENTRY(INTWDTB4TIT,	EDGE,	26, 15,  8)
+
+	ENTRY(INTDCUTDI,	EDGE,	27, 16, 16)
 
 #undef ENTRY
 };
diff --git a/include/dt-bindings/interrupt-controller/renesas-intswcd-r8a779f0.h b/include/dt-bindings/interrupt-controller/renesas-intswcd-r8a779f0.h
index 2cd5405e1375..432217f40eea 100644
--- a/include/dt-bindings/interrupt-controller/renesas-intswcd-r8a779f0.h
+++ b/include/dt-bindings/interrupt-controller/renesas-intswcd-r8a779f0.h
@@ -3,13 +3,479 @@
 #ifndef _DT_BINDINGS_INTERRUPT_CONTROLLER_RENESAS_INTSWCD_R8A779F0_H
 #define _DT_BINDINGS_INTERRUPT_CONTROLLER_RENESAS_INTSWCD_R8A779F0_H
 
-/* The below numbers must be all unique */
-
-/* RIIC */
-#define R8A779F0_INTSWCD_INTRIIC0EE	0
-#define R8A779F0_INTSWCD_INTRIIC0TEI	1
-#define R8A779F0_INTSWCD_INTRIIC0RI	2
-#define R8A779F0_INTSWCD_INTRIIC0TI	3
-#define R8A779F0_INTSWCD_INTOSTM0TINT	4
+#define R8A779F0_INTSWCD_CAN0TSOW0		0
+#define R8A779F0_INTSWCD_CAN0TSOW1		1
+#define R8A779F0_INTSWCD_CAN0TSOW2		2
+#define R8A779F0_INTSWCD_CAN0TSOW3		3
+#define R8A779F0_INTSWCD_CAN0TSOW4		4
+#define R8A779F0_INTSWCD_CAN0TSOW5		5
+#define R8A779F0_INTSWCD_CAN0TSOW6		6
+#define R8A779F0_INTSWCD_CAN0TSOW7		7
+#define R8A779F0_INTSWCD_CAN0TSIDC0		8
+#define R8A779F0_INTSWCD_CAN0TSIDC1		9
+#define R8A779F0_INTSWCD_CAN0TSIDC2		10
+#define R8A779F0_INTSWCD_CAN0TSIDC3		11
+#define R8A779F0_INTSWCD_CAN0TSIDC4		12
+#define R8A779F0_INTSWCD_CAN0TSIDC5		13
+#define R8A779F0_INTSWCD_CAN0TSIDC6		14
+#define R8A779F0_INTSWCD_CAN0TSIDC7		15
+
+#define R8A779F0_INTSWCD_PE0_INTEINTSW0		16
+#define R8A779F0_INTSWCD_PE0_INTEINTSW1		17
+#define R8A779F0_INTSWCD_PE0_INTEINTSW2		18
+#define R8A779F0_INTSWCD_PE0_INTEINTSW3		19
+
+#define R8A779F0_INTSWCD_INTICU2PES0		20
+#define R8A779F0_INTSWCD_INTICU2PE0		21
+
+#define R8A779F0_INTSWCD_PE0_INTTPTM0		22
+
+#define R8A779F0_INTSWCD_CAN1TSOW0		23
+#define R8A779F0_INTSWCD_CAN1TSOW1		24
+#define R8A779F0_INTSWCD_CAN1TSOW2		25
+#define R8A779F0_INTSWCD_CAN1TSOW3		26
+#define R8A779F0_INTSWCD_CAN1TSOW4		27
+#define R8A779F0_INTSWCD_CAN1TSOW5		28
+#define R8A779F0_INTSWCD_CAN1TSOW6		29
+#define R8A779F0_INTSWCD_CAN1TSOW7		30
+#define R8A779F0_INTSWCD_CAN1TSIDC0		31
+#define R8A779F0_INTSWCD_CAN1TSIDC1		32
+#define R8A779F0_INTSWCD_CAN1TSIDC2		33
+#define R8A779F0_INTSWCD_CAN1TSIDC3		34
+#define R8A779F0_INTSWCD_CAN1TSIDC4		35
+#define R8A779F0_INTSWCD_CAN1TSIDC5		36
+#define R8A779F0_INTSWCD_CAN1TSIDC6		37
+#define R8A779F0_INTSWCD_CAN1TSIDC7		38
+
+#define R8A779F0_INTSWCD_INTICU2PES1		39
+#define R8A779F0_INTSWCD_INTICU2PE1		40	
+
+#define R8A779F0_INTSWCD_PE1_INTTPTM1		41
+
+#define R8A779F0_INTSWCD_INTDTS31TO0		42
+#define R8A779F0_INTSWCD_INTDTS63TO32		43
+#define R8A779F0_INTSWCD_INTDTS95TO64		44
+#define R8A779F0_INTSWCD_INTDTS127TO96		45
+#define R8A779F0_INTSWCD_INTDTSCT31TO0		46
+#define R8A779F0_INTSWCD_INTDTSCT63TO32		47
+#define R8A779F0_INTSWCD_INTDTSCT95TO64		48
+#define R8A779F0_INTSWCD_INTDTSCT127TO96	49
+
+#define R8A779F0_INTSWCD_INTSDMAC0CH0		50
+#define R8A779F0_INTSWCD_INTSDMAC0CH1		51
+#define R8A779F0_INTSWCD_INTSDMAC0CH2		52
+#define R8A779F0_INTSWCD_INTSDMAC0CH3		53
+#define R8A779F0_INTSWCD_INTSDMAC0CH4		54
+#define R8A779F0_INTSWCD_INTSDMAC0CH5		55
+#define R8A779F0_INTSWCD_INTSDMAC0CH6		56
+#define R8A779F0_INTSWCD_INTSDMAC0CH7		57
+#define R8A779F0_INTSWCD_INTSDMAC0CH8		58
+#define R8A779F0_INTSWCD_INTSDMAC0CH9		59
+#define R8A779F0_INTSWCD_INTSDMAC0CH10		60
+#define R8A779F0_INTSWCD_INTSDMAC0CH11		61
+#define R8A779F0_INTSWCD_INTSDMAC0CH12		62
+#define R8A779F0_INTSWCD_INTSDMAC0CH13		63
+#define R8A779F0_INTSWCD_INTSDMAC0CH14		64
+#define R8A779F0_INTSWCD_INTSDMAC0CH15		65
+
+#define R8A779F0_INTSWCD_INTSDMAC1CH0		66
+#define R8A779F0_INTSWCD_INTSDMAC1CH1		67
+#define R8A779F0_INTSWCD_INTSDMAC1CH2		68
+#define R8A779F0_INTSWCD_INTSDMAC1CH3		69
+#define R8A779F0_INTSWCD_INTSDMAC1CH4		70
+#define R8A779F0_INTSWCD_INTSDMAC1CH5		71
+#define R8A779F0_INTSWCD_INTSDMAC1CH6		72
+#define R8A779F0_INTSWCD_INTSDMAC1CH7		73
+#define R8A779F0_INTSWCD_INTSDMAC1CH8		74
+#define R8A779F0_INTSWCD_INTSDMAC1CH9		75
+#define R8A779F0_INTSWCD_INTSDMAC1CH10		76
+#define R8A779F0_INTSWCD_INTSDMAC1CH11		77
+#define R8A779F0_INTSWCD_INTSDMAC1CH12		78
+#define R8A779F0_INTSWCD_INTSDMAC1CH13		79
+#define R8A779F0_INTSWCD_INTSDMAC1CH14		80
+#define R8A779F0_INTSWCD_INTSDMAC1CH15		81
+
+#define R8A779F0_INTSWCD_INTRCANGERR0		82
+#define R8A779F0_INTSWCD_INTRCANGRECC0		83
+#define R8A779F0_INTSWCD_INTRCAN0ERR		84
+#define R8A779F0_INTSWCD_INTRCAN0REC		85
+#define R8A779F0_INTSWCD_INTRCAN0TRX		86
+#define R8A779F0_INTSWCD_INTRCAN1ERR		87
+#define R8A779F0_INTSWCD_INTRCAN1REC		88
+#define R8A779F0_INTSWCD_INTRCAN1TRX		89
+#define R8A779F0_INTSWCD_INTRCAN2ERR		90
+#define R8A779F0_INTSWCD_INTRCAN2REC		91
+#define R8A779F0_INTSWCD_INTRCAN2TRX		92
+#define R8A779F0_INTSWCD_INTRCAN3ERR		93
+#define R8A779F0_INTSWCD_INTRCAN3REC		94
+#define R8A779F0_INTSWCD_INTRCAN3TRX		95
+#define R8A779F0_INTSWCD_INTRCAN4ERR		96
+#define R8A779F0_INTSWCD_INTRCAN4REC		97
+#define R8A779F0_INTSWCD_INTRCAN4TRX		98
+#define R8A779F0_INTSWCD_INTRCAN5ERR		99
+#define R8A779F0_INTSWCD_INTRCAN5REC		100
+#define R8A779F0_INTSWCD_INTRCAN5TRX		101
+#define R8A779F0_INTSWCD_INTRCAN6ERR		102
+#define R8A779F0_INTSWCD_INTRCAN6REC		103
+#define R8A779F0_INTSWCD_INTRCAN6TRX		104
+#define R8A779F0_INTSWCD_INTRCAN7ERR		105
+#define R8A779F0_INTSWCD_INTRCAN7REC		106
+#define R8A779F0_INTSWCD_INTRCAN7TRX		107
+
+#define R8A779F0_INTSWCD_INTRCANGERR1		108
+#define R8A779F0_INTSWCD_INTRCANGRECC1		109
+#define R8A779F0_INTSWCD_INTRCAN8ERR		110
+#define R8A779F0_INTSWCD_INTRCAN8REC		111
+#define R8A779F0_INTSWCD_INTRCAN8TRX		112
+#define R8A779F0_INTSWCD_INTRCAN9ERR		113
+#define R8A779F0_INTSWCD_INTRCAN9REC		114
+#define R8A779F0_INTSWCD_INTRCAN9TRX		115
+#define R8A779F0_INTSWCD_INTRCAN10ERR		116
+#define R8A779F0_INTSWCD_INTRCAN10REC		117
+#define R8A779F0_INTSWCD_INTRCAN10TRX		118
+#define R8A779F0_INTSWCD_INTRCAN11ERR		119
+#define R8A779F0_INTSWCD_INTRCAN11REC		120
+#define R8A779F0_INTSWCD_INTRCAN11TRX		121
+#define R8A779F0_INTSWCD_INTRCAN12ERR		122
+#define R8A779F0_INTSWCD_INTRCAN12REC		123
+#define R8A779F0_INTSWCD_INTRCAN12TRX		124
+#define R8A779F0_INTSWCD_INTRCAN13ERR		125
+#define R8A779F0_INTSWCD_INTRCAN13REC		126
+#define R8A779F0_INTSWCD_INTRCAN13TRX		127
+#define R8A779F0_INTSWCD_INTRCAN14ERR		128
+#define R8A779F0_INTSWCD_INTRCAN14REC		129
+#define R8A779F0_INTSWCD_INTRCAN14TRX		130
+#define R8A779F0_INTSWCD_INTRCAN15ERR		131
+#define R8A779F0_INTSWCD_INTRCAN15REC		132
+#define R8A779F0_INTSWCD_INTRCAN15TRX		133
+
+#define R8A779F0_INTSWCD_INTRSENT0SI		134
+#define R8A779F0_INTSWCD_INTRSENT1SI		135
+#define R8A779F0_INTSWCD_INTRSENT2SI		136
+#define R8A779F0_INTSWCD_INTRSENT3SI		137
+#define R8A779F0_INTSWCD_INTRSENT4SI		138
+#define R8A779F0_INTSWCD_INTRSENT5SI		139
+#define R8A779F0_INTSWCD_INTRSENT6SI		140
+#define R8A779F0_INTSWCD_INTRSENT7SI		141
+
+#define R8A779F0_INTSWCD_INTFXLA0LINE0		142
+#define R8A779F0_INTSWCD_INTFXLA0LINE1		143
+#define R8A779F0_INTSWCD_INTFXLA0TIM0		144
+#define R8A779F0_INTSWCD_INTFXLA0TIM1		145
+#define R8A779F0_INTSWCD_INTFXLA0TIM2		146
+#define R8A779F0_INTSWCD_INTFXLA0FDA		147
+#define R8A779F0_INTSWCD_INTFXLA0FW		148
+#define R8A779F0_INTSWCD_INTFXLA0OW		149
+#define R8A779F0_INTSWCD_INTFXLA0OT		150
+#define R8A779F0_INTSWCD_INTFXLA0IQF		151
+#define R8A779F0_INTSWCD_INTFXLA0IQE		152
+
+#define R8A779F0_INTSWCD_INTETNB0DATA		153
+#define R8A779F0_INTSWCD_INTETNB0ERR		154
+#define R8A779F0_INTSWCD_INTETNB0MNG		155
+#define R8A779F0_INTSWCD_INTETNB0MAC		156
+
+#define R8A779F0_INTSWCD_INTRIIC0EE		157
+#define R8A779F0_INTSWCD_INTRIIC0TEI		158
+
+#define R8A779F0_INTSWCD_RCAN0VM0TX		159
+#define R8A779F0_INTSWCD_RCAN0VM0RX		160
+#define R8A779F0_INTSWCD_RCAN0VM0ERR		161
+#define R8A779F0_INTSWCD_RCAN0VM1TX		162
+#define R8A779F0_INTSWCD_RCAN0VM1RX		163
+#define R8A779F0_INTSWCD_RCAN0VM1ERR		164
+#define R8A779F0_INTSWCD_RCAN0VM2TX		165
+#define R8A779F0_INTSWCD_RCAN0VM2RX		166
+#define R8A779F0_INTSWCD_RCAN0VM2ERR		167
+#define R8A779F0_INTSWCD_RCAN0VM3TX		168
+#define R8A779F0_INTSWCD_RCAN0VM3RX		169
+#define R8A779F0_INTSWCD_RCAN0VM3ERR		170
+#define R8A779F0_INTSWCD_RCAN0VM4TX		171
+#define R8A779F0_INTSWCD_RCAN0VM4RX		172
+#define R8A779F0_INTSWCD_RCAN0VM4ERR		173
+#define R8A779F0_INTSWCD_RCAN0VM5TX		174
+#define R8A779F0_INTSWCD_RCAN0VM5RX		175
+#define R8A779F0_INTSWCD_RCAN0VM5ERR		176
+#define R8A779F0_INTSWCD_RCAN0VM6TX		177
+#define R8A779F0_INTSWCD_RCAN0VM6RX		178
+#define R8A779F0_INTSWCD_RCAN0VM6ERR		179
+#define R8A779F0_INTSWCD_RCAN0VM7TX		180
+#define R8A779F0_INTSWCD_RCAN0VM7RX		181
+#define R8A779F0_INTSWCD_RCAN0VM7ERR		182
+
+#define R8A779F0_INTSWCD_RCAN1VM0TX		183
+#define R8A779F0_INTSWCD_RCAN1VM0RX		184
+#define R8A779F0_INTSWCD_RCAN1VM0ERR		185
+#define R8A779F0_INTSWCD_RCAN1VM1TX		186
+#define R8A779F0_INTSWCD_RCAN1VM1RX		187
+#define R8A779F0_INTSWCD_RCAN1VM1ERR		188
+#define R8A779F0_INTSWCD_RCAN1VM2TX		189
+#define R8A779F0_INTSWCD_RCAN1VM2RX		190
+#define R8A779F0_INTSWCD_RCAN1VM2ERR		191
+#define R8A779F0_INTSWCD_RCAN1VM3TX		192
+#define R8A779F0_INTSWCD_RCAN1VM3RX		193
+#define R8A779F0_INTSWCD_RCAN1VM3ERR		194
+#define R8A779F0_INTSWCD_RCAN1VM4TX		195
+#define R8A779F0_INTSWCD_RCAN1VM4RX		196
+#define R8A779F0_INTSWCD_RCAN1VM4ERR		197
+#define R8A779F0_INTSWCD_RCAN1VM5TX		198
+#define R8A779F0_INTSWCD_RCAN1VM5RX		199
+#define R8A779F0_INTSWCD_RCAN1VM5ERR		200
+#define R8A779F0_INTSWCD_RCAN1VM6TX		201
+#define R8A779F0_INTSWCD_RCAN1VM6RX		202
+#define R8A779F0_INTSWCD_RCAN1VM6ERR		203
+#define R8A779F0_INTSWCD_RCAN1VM7TX		204
+#define R8A779F0_INTSWCD_RCAN1VM7RX		205
+#define R8A779F0_INTSWCD_RCAN1VM7ERR		206
+
+#define R8A779F0_INTSWCD_PE0_INTECMMI		207
+#define R8A779F0_INTSWCD_PE0_INTECDCLSMI	208
+
+#define R8A779F0_INTSWCD_INTTAUD0I0		209
+#define R8A779F0_INTSWCD_INTTAUD0I2		210
+#define R8A779F0_INTSWCD_INTTAUD0I4		211
+#define R8A779F0_INTSWCD_INTTAUD0I6		212
+
+#define R8A779F0_INTSWCD_INTWDTB0TIT		213
+
+#define R8A779F0_INTSWCD_INTTAUD0I8		214
+#define R8A779F0_INTSWCD_INTTAUD0I10		215
+#define R8A779F0_INTSWCD_INTTAUD0I12		216
+#define R8A779F0_INTSWCD_INTTAUD0I14		217
+
+#define R8A779F0_INTSWCD_INTSDMACERR		218
+
+#define R8A779F0_INTSWCD_INTDTSERR		219
+
+#define R8A779F0_INTSWCD_PE1_INTECMMI		220
+#define R8A779F0_INTSWCD_PE1_INTECDCLSMI	221
+
+#define R8A779F0_INTSWCD_INTWDTB1TIT		222
+
+#define R8A779F0_INTSWCD_INTOSTM0TINT		223
+#define R8A779F0_INTSWCD_INTOSTM1TINT		224
+#define R8A779F0_INTSWCD_INTOSTM2TINT		225
+#define R8A779F0_INTSWCD_INTOSTM3TINT		226
+#define R8A779F0_INTSWCD_INTOSTM4TINT		227
+#define R8A779F0_INTSWCD_INTOSTM5TINT		228
+#define R8A779F0_INTSWCD_INTOSTM8TINT		229
+#define R8A779F0_INTSWCD_INTOSTM9TINT		230
+
+#define R8A779F0_INTSWCD_INTTPTMU00		231
+#define R8A779F0_INTSWCD_INTTPTMU01		232
+#define R8A779F0_INTSWCD_INTTPTMU02		233
+#define R8A779F0_INTSWCD_INTTPTMU03		234
+#define R8A779F0_INTSWCD_INTTPTMU10		235
+#define R8A779F0_INTSWCD_INTTPTMU11		236
+#define R8A779F0_INTSWCD_INTTPTMU12		237
+#define R8A779F0_INTSWCD_INTTPTMU13		238
+
+#define R8A779F0_INTSWCD_INTMSPI0TX0		239
+#define R8A779F0_INTSWCD_INTMSPI0RX0		240
+#define R8A779F0_INTSWCD_INTMSPI0TX1		241
+#define R8A779F0_INTSWCD_INTMSPI0RX1		242
+#define R8A779F0_INTSWCD_INTMSPI0TX2		243
+#define R8A779F0_INTSWCD_INTMSPI0RX2		244
+#define R8A779F0_INTSWCD_INTMSPI0TX		245
+#define R8A779F0_INTSWCD_INTMSPI0RX		246
+#define R8A779F0_INTSWCD_INTMSPI0FE		247
+#define R8A779F0_INTSWCD_INTMSPI0ERR		248
+
+#define R8A779F0_INTSWCD_INTMSPI1TX0		249
+#define R8A779F0_INTSWCD_INTMSPI1RX0		250
+#define R8A779F0_INTSWCD_INTMSPI1TX1		251
+#define R8A779F0_INTSWCD_INTMSPI1RX1		252
+#define R8A779F0_INTSWCD_INTMSPI1TX2		253
+#define R8A779F0_INTSWCD_INTMSPI1RX2		254
+#define R8A779F0_INTSWCD_INTMSPI1TX		255
+#define R8A779F0_INTSWCD_INTMSPI1RX		256
+#define R8A779F0_INTSWCD_INTMSPI1FE		257
+#define R8A779F0_INTSWCD_INTMSPI1ERR		258
+
+#define R8A779F0_INTSWCD_INTMSPI2TX		259
+#define R8A779F0_INTSWCD_INTMSPI2RX		260
+#define R8A779F0_INTSWCD_INTMSPI2FE		261
+#define R8A779F0_INTSWCD_INTMSPI2ERR		262
+
+#define R8A779F0_INTSWCD_INTMSPI3TX		263
+#define R8A779F0_INTSWCD_INTMSPI3RX		264
+#define R8A779F0_INTSWCD_INTMSPI3FE		265
+#define R8A779F0_INTSWCD_INTMSPI3ERR		266
+
+#define R8A779F0_INTSWCD_INTMSPI4TX		267
+#define R8A779F0_INTSWCD_INTMSPI4RX		268
+#define R8A779F0_INTSWCD_INTMSPI4FE		269
+#define R8A779F0_INTSWCD_INTMSPI4ERR		270
+
+#define R8A779F0_INTSWCD_INTMSPI5TX		271
+#define R8A779F0_INTSWCD_INTMSPI5RX		272
+#define R8A779F0_INTSWCD_INTMSPI5FE		273
+#define R8A779F0_INTSWCD_INTMSPI5ERR		274
+
+#define R8A779F0_INTSWCD_INTTAUJ1I0		275
+#define R8A779F0_INTSWCD_INTTAUJ1I1		276
+#define R8A779F0_INTSWCD_INTTAUJ1I2		277
+#define R8A779F0_INTSWCD_INTTAUJ1I3		278
+#define R8A779F0_INTSWCD_INTTAUJ3I0		279
+#define R8A779F0_INTSWCD_INTTAUJ3I1		280
+#define R8A779F0_INTSWCD_INTTAUJ3I2		281
+#define R8A779F0_INTSWCD_INTTAUJ3I3		282
+
+#define R8A779F0_INTSWCD_INTTAUD0I1		283
+#define R8A779F0_INTSWCD_INTTAUD0I3		284
+#define R8A779F0_INTSWCD_INTTAUD0I5		285
+#define R8A779F0_INTSWCD_INTTAUD0I7		286
+#define R8A779F0_INTSWCD_INTTAUD0I9		287
+#define R8A779F0_INTSWCD_INTTAUD0I11		288
+#define R8A779F0_INTSWCD_INTTAUD0I13		289
+#define R8A779F0_INTSWCD_INTTAUD0I15		290
+#define R8A779F0_INTSWCD_INTTAUD1I0		291
+#define R8A779F0_INTSWCD_INTTAUD1I1		292
+#define R8A779F0_INTSWCD_INTTAUD1I2		293
+#define R8A779F0_INTSWCD_INTTAUD1I3		294
+#define R8A779F0_INTSWCD_INTTAUD1I4		295
+#define R8A779F0_INTSWCD_INTTAUD1I5		296
+#define R8A779F0_INTSWCD_INTTAUD1I6		297
+#define R8A779F0_INTSWCD_INTTAUD1I7		298
+#define R8A779F0_INTSWCD_INTTAUD1I8		299
+#define R8A779F0_INTSWCD_INTTAUD1I9		300
+#define R8A779F0_INTSWCD_INTTAUD1I10		301
+#define R8A779F0_INTSWCD_INTTAUD1I11		302
+#define R8A779F0_INTSWCD_INTTAUD1I12		303
+#define R8A779F0_INTSWCD_INTTAUD1I13		304
+#define R8A779F0_INTSWCD_INTTAUD1I14		305
+#define R8A779F0_INTSWCD_INTTAUD1I15		306
+
+#define R8A779F0_INTSWCD_INTRLIN30		307
+#define R8A779F0_INTSWCD_INTRLIN30UR0		308
+#define R8A779F0_INTSWCD_INTRLIN30UR1		309
+#define R8A779F0_INTSWCD_INTRLIN30UR2		310
+#define R8A779F0_INTSWCD_INTRLIN31		311
+#define R8A779F0_INTSWCD_INTRLIN31UR0		312
+#define R8A779F0_INTSWCD_INTRLIN31UR1		313
+#define R8A779F0_INTSWCD_INTRLIN31UR2		314
+#define R8A779F0_INTSWCD_INTRLIN32		315
+#define R8A779F0_INTSWCD_INTRLIN32UR0		316
+#define R8A779F0_INTSWCD_INTRLIN32UR1		317
+#define R8A779F0_INTSWCD_INTRLIN32UR2		318
+#define R8A779F0_INTSWCD_INTRLIN33		319
+#define R8A779F0_INTSWCD_INTRLIN33UR0		320
+#define R8A779F0_INTSWCD_INTRLIN33UR1		321
+#define R8A779F0_INTSWCD_INTRLIN33UR2		322
+#define R8A779F0_INTSWCD_INTRLIN34		323
+#define R8A779F0_INTSWCD_INTRLIN34UR0		324
+#define R8A779F0_INTSWCD_INTRLIN34UR1		325
+#define R8A779F0_INTSWCD_INTRLIN34UR2		326
+#define R8A779F0_INTSWCD_INTRLIN35		327
+#define R8A779F0_INTSWCD_INTRLIN35UR0		328
+#define R8A779F0_INTSWCD_INTRLIN35UR1		329
+#define R8A779F0_INTSWCD_INTRLIN35UR2		330
+#define R8A779F0_INTSWCD_INTRLIN36		331
+#define R8A779F0_INTSWCD_INTRLIN36UR0		332
+#define R8A779F0_INTSWCD_INTRLIN36UR1		333
+#define R8A779F0_INTSWCD_INTRLIN36UR2		334
+#define R8A779F0_INTSWCD_INTRLIN37		335
+#define R8A779F0_INTSWCD_INTRLIN37UR0		336
+#define R8A779F0_INTSWCD_INTRLIN37UR1		337
+#define R8A779F0_INTSWCD_INTRLIN37UR2		338
+#define R8A779F0_INTSWCD_INTRLIN38		339
+#define R8A779F0_INTSWCD_INTRLIN38UR0		340
+#define R8A779F0_INTSWCD_INTRLIN38UR1		341
+#define R8A779F0_INTSWCD_INTRLIN38UR2		342
+#define R8A779F0_INTSWCD_INTRLIN39		343
+#define R8A779F0_INTSWCD_INTRLIN39UR0		344
+#define R8A779F0_INTSWCD_INTRLIN39UR1		345
+#define R8A779F0_INTSWCD_INTRLIN39UR2		346
+#define R8A779F0_INTSWCD_INTRLIN310		347
+#define R8A779F0_INTSWCD_INTRLIN310UR0		348
+#define R8A779F0_INTSWCD_INTRLIN310UR1		349
+#define R8A779F0_INTSWCD_INTRLIN310UR2		350
+#define R8A779F0_INTSWCD_INTRLIN311		351
+#define R8A779F0_INTSWCD_INTRLIN311UR0		352
+#define R8A779F0_INTSWCD_INTRLIN311UR1		353
+#define R8A779F0_INTSWCD_INTRLIN311UR2		354
+#define R8A779F0_INTSWCD_INTRLIN312		355
+#define R8A779F0_INTSWCD_INTRLIN312UR0		356
+#define R8A779F0_INTSWCD_INTRLIN312UR1		357
+#define R8A779F0_INTSWCD_INTRLIN312UR2		358
+#define R8A779F0_INTSWCD_INTRLIN313		359
+#define R8A779F0_INTSWCD_INTRLIN313UR0		360
+#define R8A779F0_INTSWCD_INTRLIN313UR1		361
+#define R8A779F0_INTSWCD_INTRLIN313UR2		362
+#define R8A779F0_INTSWCD_INTRLIN314		363
+#define R8A779F0_INTSWCD_INTRLIN314UR0		364
+#define R8A779F0_INTSWCD_INTRLIN314UR1		365
+#define R8A779F0_INTSWCD_INTRLIN314UR2		366
+#define R8A779F0_INTSWCD_INTRLIN315		367
+#define R8A779F0_INTSWCD_INTRLIN315UR0		368
+#define R8A779F0_INTSWCD_INTRLIN315UR1		369
+#define R8A779F0_INTSWCD_INTRLIN315UR2		370
+
+#define R8A779F0_INTSWCD_INTRSENT0RI		371
+#define R8A779F0_INTSWCD_INTRSENT1RI		372
+#define R8A779F0_INTSWCD_INTRSENT2RI		373
+#define R8A779F0_INTSWCD_INTRSENT3RI		374
+#define R8A779F0_INTSWCD_INTRSENT4RI		375
+#define R8A779F0_INTSWCD_INTRSENT5RI		376
+#define R8A779F0_INTSWCD_INTRSENT6RI		377
+#define R8A779F0_INTSWCD_INTRSENT7RI		378
+
+#define R8A779F0_INTSWCD_INTP00			379
+#define R8A779F0_INTSWCD_INTP01			380
+#define R8A779F0_INTSWCD_INTP02			381
+#define R8A779F0_INTSWCD_INTP03			382
+#define R8A779F0_INTSWCD_INTP04			383
+#define R8A779F0_INTSWCD_INTP05			384
+#define R8A779F0_INTSWCD_INTP06			385
+#define R8A779F0_INTSWCD_INTP07			386
+#define R8A779F0_INTSWCD_INTP08			387
+#define R8A779F0_INTSWCD_INTP09			388
+#define R8A779F0_INTSWCD_INTP10			389
+#define R8A779F0_INTSWCD_INTP11			390
+#define R8A779F0_INTSWCD_INTP12			391
+#define R8A779F0_INTSWCD_INTP13			392
+#define R8A779F0_INTSWCD_INTP14			393
+#define R8A779F0_INTSWCD_INTP15			394
+#define R8A779F0_INTSWCD_INTP16			395
+#define R8A779F0_INTSWCD_INTP17			396
+#define R8A779F0_INTSWCD_INTP18			397
+#define R8A779F0_INTSWCD_INTP19			398
+#define R8A779F0_INTSWCD_INTP20			399
+#define R8A779F0_INTSWCD_INTP21			400
+#define R8A779F0_INTSWCD_INTP22			401
+#define R8A779F0_INTSWCD_INTP23			402
+#define R8A779F0_INTSWCD_INTP24			403
+#define R8A779F0_INTSWCD_INTP25			404
+#define R8A779F0_INTSWCD_INTP26			405
+#define R8A779F0_INTSWCD_INTP27			406
+#define R8A779F0_INTSWCD_INTP28			407
+#define R8A779F0_INTSWCD_INTP29			408
+#define R8A779F0_INTSWCD_INTP30			409
+#define R8A779F0_INTSWCD_INTP31			410
+#define R8A779F0_INTSWCD_INTP32			411
+#define R8A779F0_INTSWCD_INTP33			412
+#define R8A779F0_INTSWCD_INTP34			413
+#define R8A779F0_INTSWCD_INTP35			414
+#define R8A779F0_INTSWCD_INTP36			415
+#define R8A779F0_INTSWCD_INTP37			416
+
+#define R8A779F0_INTSWCD_INTRTCA01S		417
+#define R8A779F0_INTSWCD_INTRTCA0AL		418
+#define R8A779F0_INTSWCD_INTRTCA0R		419
+
+#define R8A779F0_INTSWCD_INTRIIC0RI		420
+#define R8A779F0_INTSWCD_INTRIIC0TI		421
+
+#define R8A779F0_INTSWCD_INTDNFA1WUF0		422
+#define R8A779F0_INTSWCD_INTDNFA1WUF1		423
+#define R8A779F0_INTSWCD_INTDNFA2WUF0		424
+#define R8A779F0_INTSWCD_INTDNFA2WUF1		425
+#define R8A779F0_INTSWCD_INTDNFA0WUF0		426
+
+#define R8A779F0_INTSWCD_INTICUMFATALERR	427
+
+#define R8A779F0_INTSWCD_INTWDTB4TIT		428
+
+#define R8A779F0_INTSWCD_INTDCUTDI		429
 
 #endif
-- 
2.30.2

